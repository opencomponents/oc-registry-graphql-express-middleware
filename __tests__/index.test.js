const fetch = require('node-fetch');
const factory = require('../src/index');

const options = {
  baseUrl: 'http://mock:3000/',
  graphiql: true
};

const middleware = factory(options);

test('expect type of middleware to be function', () => {
  expect(typeof middleware).toBe('function');
});

test('expect res setHeader and end to match snapshot', async () => {
  fetch.mockResponse(JSON.stringify({
    href: options.baseUrl,
    ocVersion: '1.2.3',
    type: 'mock-registry',
    components: [
      `${options.baseUrl}oc-a-component`
    ],
    name: 'oc-a-component',
    description: 'Awesome OpenComponent',
    version: '4.5.6'
  }));

  const req = {
    method: 'GET',
    headers: {},
    url: `?query=
    {
      registry {
        href
        ocVersion
        type
      }
      components {
        name
        description
        version
      }
      component(name: "oc-a-component") {
        name
        description
        version
      }
    }
    `
  };

  const res = { setHeader: jest.fn(), end: jest.fn() };

  await middleware(req, res);

  expect(res.setHeader.mock.calls).toMatchSnapshot();
  expect(res.end.mock.calls).toMatchSnapshot();
});
