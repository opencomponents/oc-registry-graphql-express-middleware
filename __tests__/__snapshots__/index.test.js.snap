// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect res setHeader and end to match snapshot 1`] = `
Array [
  Array [
    "Content-Type",
    "application/json; charset=utf-8",
  ],
]
`;

exports[`expect res setHeader and end to match snapshot 2`] = `
Array [
  Array [
    "{\\"data\\":{\\"registry\\":{\\"href\\":\\"http://mock:3000/\\",\\"ocVersion\\":\\"1.2.3\\",\\"type\\":\\"mock-registry\\"},\\"components\\":[{\\"name\\":\\"oc-a-component\\",\\"description\\":\\"Awesome OpenComponent\\",\\"version\\":\\"4.5.6\\"}],\\"component\\":{\\"name\\":\\"oc-a-component\\",\\"description\\":\\"Awesome OpenComponent\\",\\"version\\":\\"4.5.6\\"}}}",
  ],
]
`;
